pkg/kp/healthmanager.go

```diff
@@ -269,7 +300,14 @@ func processHealthUpdater(
 			write = nil
 			if result.OK {
 				remoteHealth = result.Health
+				if result.Throttle && throttle == nil {
```

Comment:
This function has gotten quite large - what do you think about moving this logic to a `LeakyBucket{}`?